models/User.js

```diff
@@ -0,0 +1,51 @@
+const db = require('../db')
+
+class User {
+  static addUser(username, email, password) {
+    const queryText = 'INSERT INTO users (username, email, password) VALUES ($1, $2, $3)'
+    return db.query(queryText, [username, email, password])
+  }
+
+  static getByEmail (email) {
+    const queryText = 'SELECT * FROM users WHERE email = $1'
+    return db.query(queryText, [email])
+      .then((data) => data.rows[0])
+  }
+
+  static getById (userId) {
+    const queryText = 'SELECT * FROM users WHERE id = $1'
+    return db.query(queryText, [userId])
+  }
+
+  static getByUsername (username) {
+    const queryText = 'SELECT * FROM users WHERE username = $1;'
+    return db.query(queryText, [username])
+      .then((data) => data.rows[0])
+  }
+
+  static getAllUsers () {
+    return db.query('SELECT * FROM users;')
```

Comment:
Can't think of the use for this one for sprint 1 but since you have it, consider returning just the array of results here to keep it consistent with what other methods return